storage-resize-images/functions/src/index.ts

```diff
@@ -49,149 +47,194 @@ logs.init();
-              size,
-              objectMetadata: objectMetadata,
-              format,
-            })
-          );
-        });
+    const bucket = admin.storage().bucket(process.env.IMG_BUCKET);
+    const query = data.nextPageQuery || {
+      autoPaginate: false,
+      maxResults: 3,
+    };
+    const [files, nextPageQuery] = await bucket.getFiles(query);
+    const filesToResize = files.filter((f) => {
+      logs.continueBackfill(f.metadata.name);
+      return shouldResize(f.metadata);
+    });
+    const filePromises = filesToResize.map((f) => {
+      return generateResizedImageHandler(f.metadata, false);
+    });
+    const results = await Promise.allSettled(filePromises);
+
+    const pageErrorsCount = results.filter(
+      (r) => r.status === "rejected"
+    ).length;
+    const pageSuccessCount = results.filter(
+      (r) => r.status === "fulfilled"
+    ).length;
+    const oldErrorsCount = Number(data.errorsCount) || 0;
+    const oldSuccessCount = Number(data.successCount) || 0;
+    const errorsCount = pageErrorsCount + oldErrorsCount;
+    const successCount = pageSuccessCount + oldSuccessCount;
+
+    if (nextPageQuery) {
+      const queue = getFunctions().taskQueue(
+        `backfillResizedImages`,
+        process.env.EXT_INSTANCE_ID
+      );
+      await queue.enqueue({
+        nextPageQuery,
+        errorsCount,
+        successCount,
       });
-
-      const results = await Promise.all(tasks);
-      eventChannel &&
-        (await eventChannel.publish({
-          type: "firebase.extensions.storage-resize-images.v1.complete",
-          subject: filePath,
-          data: {
-            input: object,
-            outputs: results,
-          },
-        }));
-
-      const failed = results.some((result) => result.success === false);
-      if (failed) {
-        logs.failed();
-        return;
-      } else {
-        if (config.deleteOriginalFile === deleteImage.onSuccess) {
-          if (remoteFile) {
-            try {
-              logs.remoteFileDeleting(filePath);
-              await remoteFile.delete();
-              logs.remoteFileDeleted(filePath);
-            } catch (err) {
-              logs.errorDeleting(err);
-            }
-          }
-        }
-        logs.complete();
-      }
-    } catch (err) {
-      logs.error(err);
-    } finally {
-      if (originalFile) {
-        logs.tempOriginalFileDeleting(filePath);
-        fs.unlinkSync(originalFile);
-        logs.tempOriginalFileDeleted(filePath);
+    } else {
+      logs.backfillComplete(successCount, errorsCount);
+      if (errorsCount == 0) {
+        await runtime.setProcessingState(
+          "PROCESSING_COMPLETE",
+          `Successfully resized ${successCount} images.`
+        );
+      } else if (errorsCount > 0 && successCount > 0) {
+        await runtime.setProcessingState(
+          "PROCESSING_WARNING",
+          `Successfully resized ${successCount} images, failed to resize ${errorsCount} images.`
```

Comment:
Are the failures in the logs? If so, we should probably mention that here. If not, we should probably put them in the logs and then mention that here :)